mqtt_client.c

```diff
@@ -957,6 +957,14 @@ esp_err_t esp_mqtt_client_reconnect(esp_mqtt_client_handle_t client)
 
 esp_err_t esp_mqtt_client_stop(esp_mqtt_client_handle_t client)
 {
+    // Notify the broker we are disconnecting
```

Comment:
This change could cause a crash if `esp_mqtt_client_stop` is called if client not started. I would suggest to move under `if (client->run)` statement.